src/actions/actions.ts

```diff
@@ -0,0 +1,66 @@
+import { Webhooks } from '../webhooks/webhooks';
+import { ResponsePayload } from './interfaces/response-payload';
+
+export class Actions {
+  private webhooks: Webhooks;
+
+  constructor(webhooks: Webhooks) {
+    this.webhooks = webhooks;
+  }
+
+  private get computeSignature() {
+    return this.webhooks.computeSignature.bind(this.webhooks);
+  }
+
+  get verifyHeader() {
+    return this.webhooks.verifyHeader.bind(this.webhooks);
+  }
+
+  async signResponse({
+    type,
+    verdict,
+    errorMessage,
+    secret,
+  }: {
+    type: 'authentication' | 'user_registration';
+    verdict: 'allow' | 'deny';
+    secret: string;
+    errorMessage?: string;
+  }) {
+    const responsePayload: ResponsePayload = {
+      timestamp: Date.now(),
+      verdict,
+      ...(errorMessage && { error_message: errorMessage }),
+    };
+
+    if (verdict === 'allow' && errorMessage) {
+      throw new Error('Cannot have an error message with a verdict of allow');
+    }
```

Comment:
If we change the typing like we do above this becomes less valuable... thought I guess it's still useful for anyone using plain old JS and type TypeScript?